### ISSUE TITLE
Using numeric escape for backslash does not work in string literal

### ISSUE THREAD

**Issue by rdulmina** [ballerina-platform/ballerina-lang#33051]
**Description:**
Consider the bellow sample
```bal
function init() {
    string c = "A \u{5C} B";
    io:println(c); // @output A \ B
}
```

This should print `A \ B` instead of  `A  B` 

**Affected Versions:**
Current master

**Comment by pcnfernando** [comment]
Getting below in SL beta 3 rc5

![Screenshot 2021-10-06 at 15 44 39](https://user-images.githubusercontent.com/15815199/136184265-c9aaa8e3-1775-4022-b1dc-79c555e9d6e5.png)

**Comment by rdulmina** [comment]
Sorry, the example should be updated with `5C`. will update

**Comment by rdulmina** [comment]
> I think I'm still not getting the issue here. `"A \u{5C} B"` should be unescaped to `A \ B` right?

Yes, the thing is it's not working on the current master

**Comment by lochana-chathura** [comment]
```bal
// Beta2 output
io:println("A \u{5C} B"); // prints A \ B
io:println("A \u{5C} B" == "A \\ B"); // prints true

// Beta3 output
io:println("A \u{5C} B"); // prints A  B
io:println("A \u{5C} B" == "A \\ B"); // prints false
```

**Comment by rdulmina** [comment]
Fixed with #33302
